File: materials/OOP/attribute_object/C++/student_details/student_details.cpp
#include "student_details_def.hpp"

StudentDetails::StudentDetails (std::string Name, int Age, std::string University) {

  this->Name = Name;
  this->Age = Age;
  this->University = University;
}

StudentDetails::~StudentDetails () {
  // pass
}

StudentDataQuery::StudentDataQuery () {
  // Pass
}

void StudentDataQuery::ShowStudentData (StudentDetails StudentObject) {

  LOG_DATA (StudentObject.Name);
  NEWLINE ();
  LOG_DATA (StudentObject.Age);
  NEWLINE ();
  LOG_DATA (StudentObject.University);
  NEWLINE ();
}

std::string StudentDataQuery::GetStudentName (StudentDetails StudentObject) {

  return StudentObject.Name;
}

int StudentDataQuery::GetStudentAge (StudentDetails StudentObject) {

  return StudentObject.Age;
}

std::string StudentDataQuery::GetStudentUniversity (StudentDetails StudentObject) {

  return StudentObject.University;
}

StudentDataQuery::~StudentDataQuery () {
  // pass
}

StudentDataManipulation::StudentDataManipulation () {
  // pass
}

void StudentDataManipulation::ChangeStudentName(StudentDetails & StudentObject, std::string NewName) {

  StudentObject.Name = NewName;
}

void StudentDataManipulation::ChangeStudentAge (StudentDetails & StudentObject, int NewAge) {

  StudentObject.Age = NewAge;
}

void StudentDataManipulation::ChangeStudentUniversity (StudentDetails & StudentObject, std::string NewUniversity) {

  StudentObject.University = NewUniversity;
}

StudentDataManipulation::~StudentDataManipulation () {
  // pass
}

File: materials/OOP/attribute_object/Java/src/com/company/student_details.java
package com.company;

class StudentDetails {

    public String Name;
    public int Age;
    public String University;

    public StudentDetails(String NameParameter, int AgeParameter, String UniversityParameter) {

        Name = NameParameter;
        Age = AgeParameter;
        University = UniversityParameter;
    }
}

class StudentDataQuery {

    public static void ShowStudentData(StudentDetails Details) {

        System.out.println(Details.Name);
        System.out.println(Details.Age);
        System.out.println(Details.University);
    }

    public static String GetStudentName(StudentDetails Details) {

        return Details.Name;
    }

    public static int GetStudentAge(StudentDetails Details) {

        return Details.Age;
    }

    public static String GetStudentUniversity(StudentDetails Details) {

        return Details.University;
    }
}

class StudentDataManipulation {

    public static void ChangeStudentName(StudentDetails Details, String NewName) {

        Details.Name = NewName;
    }

    public static void ChangeStudentAge(StudentDetails Details, int NewAge) {

        Details.Age = NewAge;
    }

    public static void ChangeStudentUniversity(StudentDetails Details, String NewUniversity) {

        Details.University = NewUniversity;
    }
}
